### ISSUE TITLE
Ignore trailing whitespaces in comments [Layout/TrailingWhiteSpace]

### ISSUE THREAD

**Issue by thestelz** [rubocop/rubocop#10214]
## Is your feature request related to a problem? Please describe.

I don't believe this is an issue, but could be mistaken. When we create documentation for a method (following YARD), RuboCop will complain about a trailing whitespace in those comments, but we want those to be ignored.

I have checked the [configuration settings](https://docs.rubocop.org/rubocop/1.22/cops_layout.html#configurable-attributes-54) for [Layout/TrailingWhitespace](https://docs.rubocop.org/rubocop/1.22/cops_layout.html#layouttrailingwhitespace) and I don't see any way to shut this off.

## Describe the solution you'd like

I would like a configuration setting added that will ignore trailing whitespace in comments (disabled by default to keep things the way they are for everyone).

## Describe alternatives you've considered

To manually delete trailing whitespaces in code comments as we find them, but this is a pain. 

## Additional context

This has come up because my company has recently added RuboCop into our established codebase, so we are having failures due to this issue.

```shell
% rubocop -v
1.22.2
```

I'm thinking a setting like below would work:

```yaml
Layout/TrailingWhiteSpace:
  AllowInComments: true|false (default)
```

<img width="568" alt="Example" src="https://user-images.githubusercontent.com/1774502/138883537-06b73add-b2fd-4f59-ad9c-f375cc76f63d.png">

I would put in a pull request for this issue myself, but I'm not familiar with your code base, so I'm not sure where to begin. If someone could point me in the right direction and this is approved, I could try and figure it out.

Thanks

**Comment by ThomasOwens** [comment]
@dvandersluis I'm not sure what Rubocop's policies are, but is it possible to reconsider this?

I just ran into this same issue where trailing whitespace in comments can't be ignored, but it is possible to ignore them in HEREDOCs. I understand that removing the trailing whitespace would not have changed the appearance of the comments, but they were there because part of the comment was a table and it represented an empty cell. Sure, there are workarounds (like putting a | at the end of the row or something), but it seems like AllowInComments would be a good companion to AllowInHeredoc.

I'd be interested in looking into this myself, if I could know that a possible pull request wouldn't be rejected outright. Otherwise, I'll just stick to disabling this cop entirely. I've found the disabling of the cop in a block of code to be clumsy and noisy.
